quantum/pointing_device/pointing_device_scroll.c

```diff
@@ -0,0 +1,241 @@
+/* Copyright 2022 Alabastard
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef POINTING_DEVICE_SCROLL_ENABLE
+
+#    include "pointing_device_scroll.h"
+
+/* Scrolling Functions and Variables */
+static scroll_record_t local_scroll_record = {.mode_tg = SM_NONE}; // store scroll mode and accumulated scroll values
```

Comment:
Would be nice to have something like `.mode_tg = POINTING_DEVICE_SCROLL_DEFAULT_MODE` if user wants to have scroll as primary function?

Reply:
yeah that is a good idea!
Also been meaning to ask do you have a keyboard with pointing devices on both hands I'm wondering if there should be some handling of split pointing devices for this feature so it effects one of the two sides (or both?)

Reply:
Yeah. This is starting to be more common nowadays. I myself have about 7 splits with pointing devices on both sides.

Reply:
You're right, there should be some handling for split use case with pointing device on both sides.
I currently don't have pointing device on my left half, been meaning to add it for testing because I keep forgetting about this use case without an example in front of me.
Ideally this feature would be controlled independently for each side; that would require double the keycodes for both sides?
Letting user specify which side(s) to scroll is much simpler to implement in comparison, but I wonder if it's too limiting. Perhaps allowing user to switch sides at runtime is good enough.

Reply:
I think run time control of which side is being hijacked would be best will need specific handling depending on if `POINTING_DEVICE_COMBINED`, `POINTING_DEVICE_LEFT` or `POINTING_DEVICE_RIGHT` is defined I suppose.